core/domain/user_services.py

```diff
@@ -443,11 +444,12 @@ def get_user_settings_by_auth_id(
         return None
 
 
-def get_user_roles_from_id(user_id: str) -> List[str]:
+def get_user_roles_from_id(user_id: Optional[str]) -> List[str]:
```

Comment:
Ditto, please try not to make things optional. Handle the None case in the caller. I think I only gave approval for get_user_actions_info() but not the other functions, and in general we should try to keep the interface constrained to reduce the "surface area".

Reply:
Fixed!